### PR TITLE
Add support for a config.php file

### PR DESCRIPTION
**Pull request by giorgiosironi**
Still allows to fall back to a environment-specific file.

Removes the `local` environment as it's a mess to setup and maintain, if we want something different from a VM we should look into a set of containers.

### CODE REVIEW

**Comment by giorgiosironi** [comment]
Could do much more, but don't wanna introduce conflicts.

File: src/Search/Console.php [hunk 1]
```diff
@@ -210,7 +210,7 @@ public function __construct(Application $console, Kernel $app)
         // Some annotations
         Register::registerLoader();
 
-        $this->console->getDefinition()->addOption(new InputOption('--env', '-e', InputOption::VALUE_REQUIRED, 'The Environment name.', 'dev'));
+        $this->console->getDefinition()->addOption(new InputOption('--env', '-e', InputOption::VALUE_OPTIONAL, 'The Environment name. Deprecated and not used', 'dev'));
```
Comment (thewilkybarkid) [inline src/Search/Console.php]: Guess setting it causes a problem?

File: bin/console [hunk 2]
```diff
@@ -8,18 +8,10 @@ use Symfony\Component\Console\Application;
 use Symfony\Component\Console\Input\ArgvInput;
 use Symfony\Component\Console\Output\ConsoleOutput;
 
-require_once __DIR__.'/../vendor/autoload.php';
-
-umask(002);
+require_once __DIR__.'/../bootstrap.php';
```
Comment (thewilkybarkid) [inline bin/console]: Move the bootstrap into `src`?

File: bootstrap.php [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+require_once __DIR__.'/vendor/autoload.php';
+umask(0002);
+$defaultFile = __DIR__.'/config/config.php';
+$legacyFile = __DIR__.'/config/'.getenv('ENVIRONMENT_NAME').'.php';
+if (file_exists($defaultFile)) {
+    $config = include $defaultFile;
```
Comment (thewilkybarkid) [inline bootstrap.php]: Take a look at https://github.com/elifesciences/recommendations/pull/100 where I'm working on a slightly different approach.

File: src/Search/Console.php (hunk 1 above)
Comment (giorgiosironi) [reply src/Search/Console.php]: Could not cause a problem, there is is some wrangling of `--env` in `bin/console`, so put this as a precaution to remove it down the line.

File: bin/console (hunk 2 above)
Comment (giorgiosironi) [reply bin/console]: Can do

File: bootstrap.php (hunk 3 above)
Comment (giorgiosironi) [reply bootstrap.php]: Should be easy to add the environment-specific files, merge them, move the main one up a folder

**Comment by giorgiosironi** [comment]
Done, tests (for now) pass.

**Pull request merged**